### ISSUE TITLE
Can you give a boundary loss for binary segmentation, thanks

### ISSUE THREAD

**Issue by ustczhouyu** [yiskw713/boundary_loss_for_remote_sensing#2]
Can you give a boundary loss for binary segmentation, or can you tell me how to modify the code to use it for binary segmentation， thanks

**Comment by yiskw713** [comment]
Hi, ustczhouyu

Sorry for the delay in replying.
How about converting binary prediction to two-class prediction?
```
import torch
n_batch = 4
H = 112
W = 112

# prediction for binary cross entropy
binary_pred = torch.rand(n_batch, 1, H, W)

# background prediction
bg_pred = torch.ones_like(binary) - binary_pred

# foreground prediction
fg_pred = binary_pred.clone()

# 2 class prediction. shape => (n_batch, 2, H, W)
pred = torch.cat([bg_pred, fg_pred], dim=1)
```

**Comment by NguyenHungHust256** [comment]
> Hi, ustczhouyu
> 
> Sorry for the delay in replying.
> How about converting binary prediction to two-class prediction?
> 
> ```
> import torch
> n_batch = 4
> H = 112
> W = 112
> 
> # prediction for binary cross entropy
> binary_pred = torch.rand(n_batch, 1, H, W)
> 
> # background prediction
> bg_pred = torch.ones_like(binary) - binary_pred
> 
> # foreground prediction
> fg_pred = binary_pred.clone()
> 
> # 2 class prediction. shape => (n_batch, 2, H, W)
> pred = torch.cat([bg_pred, fg_pred], dim=1)
> ```

Can you explain code. I don't know why you implement this code

**Comment by ustczhouyu** [comment]
Thanks for sharing your code. When doing binary segmentation (one class is the object, the other class is the background), I want to know this function: loss = criterion(y['out'], gt), where 'gt' should be the ground-truth of  the whole instance or only ground-truth of the boundary?  @yiskw713

**Comment by ustczhouyu** [comment]
Thanks for sharing your code. When doing binary segmentation (one class is the object, the other class is the background), I want to know this function: loss = criterion(y['out'], gt), where 'gt' should be the ground-truth of  the whole instance or only ground-truth of the boundary?  @yiskw713

**Comment by ustczhouyu** [comment]
Thanks for sharing your code. When doing binary segmentation (one class is the object, the other class is the background), I want to know this function: loss = criterion(y['out'], gt), where 'gt' should be the ground-truth of  the whole instance or only ground-truth of the boundary?  @yiskw713

**Comment by ustczhouyu** [comment]
Thanks for sharing your code. When doing binary segmentation (one class is the object, the other class is the background), I want to know this function: loss = criterion(y['out'], gt), where 'gt' should be the ground-truth of  the whole instance or only ground-truth of the boundary?  @yiskw713

**Comment by ustczhouyu** [comment]
Thanks for sharing your code. When doing binary segmentation (one class is the object, the other class is the background), I want to know this function: loss = criterion(y['out'], gt), where 'gt' should be the ground-truth of  the whole instance or only ground-truth of the boundary?  @yiskw713

**Comment by ustczhouyu** [comment]
Thanks for sharing your code. When doing binary segmentation (one class is the object, the other class is the background), I want to know this function: loss = criterion(y['out'], gt), where 'gt' should be the ground-truth of  the whole instance or only ground-truth of the boundary?  @yiskw713

**Comment by ustczhouyu** [comment]
Thanks for sharing your code. When doing binary segmentation (one class is the object, the other class is the background), I want to know this function: loss = criterion(y['out'], gt), where 'gt' should be the ground-truth of  the whole instance or only ground-truth of the boundary?  @yiskw713
